internal/provider/fake_server_test.go

```diff
@@ -0,0 +1,251 @@
+	require.NoError(s.t, err)
+
+	s.url = l.Addr().String()
+	s.server = grpc.NewServer()
+	cloudexportpb.RegisterCloudExportAdminServiceServer(s.server, s)
+
+	go func() {
+		err = s.server.Serve(l)
+		assert.NoError(s.t, err)
+		s.done <- struct{}{}
+	}()
+}
+
+// Stop blocks until the server is stopped.
+func (s *fakeCloudExportServer) Stop() {
+	s.server.GracefulStop()
+	<-s.done
+}
+
+// URL returns the server URL.
+func (s *fakeCloudExportServer) URL() string {
+	return fmt.Sprintf("http://%v", s.url)
+}
+
+func (s *fakeCloudExportServer) ListCloudExport(
+	_ context.Context, _ *cloudexportpb.ListCloudExportRequest,
+) (*cloudexportpb.ListCloudExportResponse, error) {
+	return &cloudexportpb.ListCloudExportResponse{
+		Exports:             s.data,
+		InvalidExportsCount: 0,
+	}, nil
+}
+
+func (s *fakeCloudExportServer) GetCloudExport(
+	ctx context.Context, req *cloudexportpb.GetCloudExportRequest,
+) (*cloudexportpb.GetCloudExportResponse, error) {
+	if idx := s.findByID(req.GetId()); idx != cloudExportNotFound {
+		return &cloudexportpb.GetCloudExportResponse{Export: s.data[idx]}, nil
+	}
+	return nil, fmt.Errorf("cloud export with ID %q not found", req.GetId())
+}
+
+func (s *fakeCloudExportServer) CreateCloudExport(
+	ctx context.Context, req *cloudexportpb.CreateCloudExportRequest,
+) (*cloudexportpb.CreateCloudExportResponse, error) {
+	newExport := req.GetExport()
+
+	if s.findByName(newExport.Name) != cloudExportNotFound {
+		return nil, fmt.Errorf("cloud export %q already exists", newExport.Name)
+	}
+
+	newExport.Id = s.allocateNewID()
+	newExport.CurrentStatus = &cloudexportpb.Status{
+		Status:               "OK",
+		ErrorMessage:         "No errors",
+		FlowFound:            &wrapperspb.BoolValue{Value: true},
+		ApiAccess:            &wrapperspb.BoolValue{Value: true},
+		StorageAccountAccess: &wrapperspb.BoolValue{Value: true},
+	}
+
+	s.data = append(s.data, newExport)
+
+	return &cloudexportpb.CreateCloudExportResponse{
+		Export: newExport,
+	}, nil
+}
+
+func (s *fakeCloudExportServer) UpdateCloudExport(
+	ctx context.Context, req *cloudexportpb.UpdateCloudExportRequest,
+) (*cloudexportpb.UpdateCloudExportResponse, error) {
+	exportUpdate := req.GetExport()
+	if i := s.findByID(exportUpdate.GetId()); i != cloudExportNotFound {
+		s.data[i] = exportUpdate
+		return &cloudexportpb.UpdateCloudExportResponse{
+			Export: exportUpdate,
+		}, nil
+	}
+	return nil, fmt.Errorf("cloud export of id %q doesn't exists", exportUpdate.Id)
+}
+
+func (s *fakeCloudExportServer) DeleteCloudExport(
+	ctx context.Context, req *cloudexportpb.DeleteCloudExportRequest,
+) (*cloudexportpb.DeleteCloudExportResponse, error) {
+	return &cloudexportpb.DeleteCloudExportResponse{}, nil
+}
+
+func (s *fakeCloudExportServer) allocateNewID() string {
+	var id int
+
+	for _, item := range s.data {
+		itemID, err := strconv.Atoi(item.Id)
+		if err != nil {
+			itemID = 1000000 // str conversion error, assume some high integer for the id
```

Comment:
Why is this needed? Can it even work? I'd prefer to print a descriptive error message and terminate if the `item.Id` string is not a valid integer.